src/lib/components/Tab/index.tsx

```diff
@@ -37,60 +49,110 @@ export const tabItemStyleClasses: Record<TabStyle, Record<TabItemStatus, string>
+    ),
+  );
+
+  const handleKeyboard = (event: KeyboardEvent): void => {
+    let targetTab = parseInt(event.currentTarget?.id.replace('tab-', ''));
+
+    if (event.key === 'ArrowLeft') {
+      setFocusedTab(Math.max(0, focusedTab - 1));
+    }
+
+    if (event.key === 'ArrowRight') {
+      setFocusedTab(Math.min(tabs.length - 1, focusedTab + 1));
+    }
+
+    if (event.key === 'Enter') {
+      targetTab = targetTab === focusedTab ? targetTab : focusedTab;
+      setActiveTab(targetTab);
+      setFocusedTab(targetTab);
+    }
+  };
+
+  useEffect(() => {
+    const focusedButton = document.getElementById(`${idPrefix}-tab-${focusedTab}`) as HTMLButtonElement;
+    focusedButton?.focus();
+  }, [focusedTab, idPrefix]);
 
   return (
     <div className="flex flex-col gap-2">
       <div
+        aria-label="Tabs"
+        role="tablist"
         className={classNames('flex text-center', tabGroupStyleClasses[style], {
           'border-b border-gray-200 dark:border-gray-700': style === 'underline',
         })}
+        {...rest}
       >
         {tabs.map((tab, index) => (
           <button
             key={index}
+            aria-controls={`${idPrefix}-panel-${index}`}
+            aria-selected={index === activeTab}
             className={classNames(
               'flex items-center justify-center p-4 text-sm font-medium first:ml-0 disabled:cursor-not-allowed disabled:text-gray-400 disabled:dark:text-gray-500',
               {
                 'ml-2 first:ml-0': style !== 'fullWidth',
                 'rounded-t-lg': style === 'underline' || style === 'default',
                 'w-full first:rounded-l-lg last:rounded-r-lg': style === 'fullWidth',
-                [tabItemStyleClasses[style].active]: index === activeTabIndex,
-                [tabItemStyleClasses[style].notActive]: index !== activeTabIndex && !tab.disabled,
+                [tabItemStyleClasses[style].active]: index === activeTab,
+                [tabItemStyleClasses[style].notActive]: index !== activeTab && !tab.disabled,
               },
             )}
-            onClick={() => setActiveTabIndex(index)}
             disabled={tab.disabled}
+            id={`${idPrefix}-tab-${index}`}
+            onClick={() => {
+              setActiveTab(index);
+              setFocusedTab(index);
+            }}
+            onKeyDown={handleKeyboard}
+            role="tab"
+            tabIndex={index === focusedTab ? 0 : -1}
           >
             {tab.icon && <tab.icon className="mr-2 h-5 w-5" />}
             {tab.title}
           </button>
         ))}
       </div>
-      <div className={classNames('p-4', className, tabs[activeTabIndex].className)}>{tabs[activeTabIndex].content}</div>
+      <div>
+        {tabs.map((tab, index) => (
+          <div
+            key={index}
+            aria-labelledby={`${idPrefix}-tab-${index}`}
+            className={classNames('p-4', className, tabs[activeTab].className)}
+            hidden={index !== activeTab}
```

Comment:
Shouldn't this be a classname?

Reply:
Haha, that's what I thought too. It turns out there's actually native support for a `hidden` attribute: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/hidden It's recommended in the MDN example on using tabs.